main.go

```diff
@@ -5,53 +5,128 @@ Copyright © 2022 Asdrubal Gonzalez Penton agpenton@gmail.com
 package main
 
 import (
-	"bufio"
 	"fmt"
+	"log"
 	"os"
+	"runtime/debug"
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws/session"
 )
 
 func check(e error) {
+	if e != nil {
+		log.Println(e)
+		debug.PrintStack()
+	}
+}
+
+func checkPanic(e error) {
 	if e != nil {
 		panic(e)
+		debug.PrintStack()
+	}
+}
+
+func checkFatal(e error) {
+	if e != nil {
+		log.Fatal(e)
+		debug.PrintStack()
+	}
+}
+
+func reportPanic() {
+	p := recover()
+	if p == nil {
+		return
+	}
+	err, ok := p.(error)
+	if ok {
+		fmt.Println(err)
+	} else {
+		panic(p)
 	}
 }
```

Comment:
i would move such methods out of the `main.go` file, so it actually only contains the code that starts the application.